src/Symfony/Component/HttpKernel/Profiler/FileProfilerStorage.php

```diff
@@ -194,11 +196,55 @@ public function write(Profile $profile): bool
                 $profile->getStatusCode(),
             ]);
             fclose($file);
+
+            $this->removeExpiredProfiles($indexFilename);
```

Comment:
perhaps play this conditionaly like one every 50 request?
otherwise, each write profile will have also this overhead no?

Reply:
I don't this we care about the overhead here, but doing it only every 50 requests on average sounds good to me.

Reply:
How can I achieve this ? Saving profile is done `onKernelResponse` and not care about other requests.

Reply:
Of course no context between request
My point here is to only execute this function which ultimatly remove old profiles only on a random basic, like 1/50 request
Its not like we handle a precise counter and every 50 we remove, its just to reduce the i/o disc not for all requests